File: 2884.c
#include<stdio.h>

int main(){
    int inHour, inMinute, outHour, outMinute;
    scanf("%d %d", &inHour, &inMinute);
    if(inMinute < 45){
        outMinute = 15 + inMinute;
        outHour = inHour - 1;
    }
    else{
        outMinute = inMinute - 45;
        outHour = inHour;
    }
    
    if(outHour < 0){
        outHour = 24 + outHour;
    }
    
    printf("%d %d", outHour, outMinute);
    return 0;
}

File: 2884.py
inHour, inMinute = map(int, input().split())
outHour = 0
outMinute = 0
if inMinute < 45 :
    outMinute = 15 + inMinute
    outHour = inHour - 1
else :
    outMinute = inMinute - 45
    outHour = inHour
if outHour < 0 :
    outHour = 24 + outHour
print(outHour,outMinute)                #AUTOMATICALLY INSERT SPACE BETWEEN VARIABLES WHEN USING , BETWEEN
